MirrorProvider/MirrorProvider.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -164,6 +166,16 @@ private void OnNewFileCreated(string relativePath, bool isDirectory)
         {
             Console.WriteLine($"OnNewFileCreated (isDirectory: {isDirectory}): {relativePath}");
         }
+
+        private void OnFileRenamed(string relativeDestinationPath, bool isDirectory)

```

Comment:
It seems really odd to get a rename event with only a single path in it. I understand the reasons why, but it's confusing from an API perspective. Should we just treat this as a new file event instead, and just not have a rename event?

Reply:
For historical purposes, it should be noted that we *could* deliver both paths on Mac, but because VFSForGit does not strictly need the source path on Mac we opted to keep the kext simpler and only deliver the destination path for now.

>  but it's confusing from an API perspective. Should we just treat this as a new file event instead, and just not have a rename event?

I've found it helpful when debugging\developing to keep these as separate events (and Keeping them as separate events also makes us more consistent with ProjFS on Windows).

I lean toward keeping them separate and having VFSForGit decide to handle this notification in the same way as a new file notification.

Reply:
I've added some additional comments in the notification callback